samples/bluetooth/mesh/light_dimmer/src/model_handler.c

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+/**
+ * @file
+ * @brief Model handler for the Light Dimmer Controller.
+ *
+ */
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/mesh/proxy.h>
+#include <bluetooth/mesh/models.h>
+#include <dk_buttons_and_leds.h>
+#include "model_handler.h"
+#include <time.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(dimmer_sample, LOG_LEVEL_INF);
+
+#define SHORT_PRESS_THRESHOLD_MS 300
+#define LONG_PRESS_TIMEOUT_SEC K_SECONDS(20)
+#define DELTA_MOVE_STEP 1000
+
+enum button_state_t { BUTTON_STATE_RELEASED, BUTTON_STATE_PRESSED };
+
+/** Context for a single Light Dimmer instance. */
+static struct dim_button {
+	/** Generic Level client instance for this switch. */
+	struct bt_mesh_lvl_cli client;
+	/* Relative start time of the latest press of the button */
+	int64_t press_start_time;
+	/* Button state: can be "pressed" or "released" */
+	enum button_state_t button_state;
+	/* Toggle state: 0: down/off 1: up/on */
+	bool tgl_state;
```

Comment:
```suggestion
	int direction;
```

and then 

```
enum dim_direction { DIM_DOWN, DIM_UP };
```

Reply:
Should it be `enum dim_direction direction;` then?

Reply:
`buttons[n].direction = buttons[n].direction == DIM_UP ? DIM_DOWN : DIM_UP;`

Makes the intent very clear, I think.